docs/gcp-openshift.md

```diff
@@ -0,0 +1,160 @@
+# Use external on Openshift in GCP environment
+
+### This document provides information about how to use the `ExternalDNS Operator` in OpenShift Container Platform on GCP.
+
+### Steps
+1. Export your kubeconfig
+```bash
+ $ export KUBECONFIG=~/<path to>/kubeconfig
+```
+
+2. Check user. The user shall access to kube-system.
+
+```bash
+ $ oc whoami
+ system:admin
+```
+
+3. Copy the value of service_account.json in gcp-credentials secret in a file encoded-gcloud.json
+```bash
+$ oc get secret gcp-credentials -n kube-system -o yaml
+```
+
+4. Decode that string to file called deocoded-gcloud.json
+```bash
+cat encoded-gcloud.json | base64 -d > decoded-gcloud.json
+```
+
+5. Export Google credentials
+```bash
+$ export GOOGLE_CREDENTIALS=decoded-gcloud.json
+```
+
+6. Activate your account
+```bash
+gcloud auth activate-service-account  <client_email as per decoded-gcloud.json> --key-file=gcp-config-decoded.json
+```
+
+7. Set your project
+```bash
+ gcloud config set project <project_id as per decoded-gcloud.json>
+```
+
+8. Get the routes to check the domain
+```bash
+oc get routes --all-namespaces | grep console
+openshift-console          console             console-openshift-console.apps.misalunk-azure.qe.azure.devcluster.openshift.com                       console             https   reencrypt/Redirect     None
+openshift-console          downloads           downloads-openshift-console.apps.misalunk-azure.qe.azure.devcluster.openshift.com                     downloads           http    edge/Redirect          None
+```
+
+9. Get your zone
+```bash
+$ gcloud dns managed-zones list | grep <dns name eg- As per the route the section after apps. i.e misalunk-azure.qe.azure.devcluster.openshift.com>
+```
+
+10. Check dns records
+```bash
+$ gcloud dns record-sets list --zone=<the zone name you get from step 9>
+```
+
+11. Create a CR
+```yaml
+
+  apiVersion: externaldns.olm.openshift.io/v1alpha1
+  kind: ExternalDNS
+  metadata:
+    name: sample-gcp
+  spec:
+    domains:
+    - filterType: Include
+      matchType: Exact
+      name: misalunkgcp.qe.gcp.devcluster.openshift.com
+    provider:
+      type: GCP
```

Comment:
In this doc https://github.com/openshift/external-dns-operator/blob/main/docs/usage.md#gcp it seems gcp `.credentials` and `.project` are required, but they are not needed here?

Reply:
No cco will get the credentials in OCP for aws, azure and gcp. The code has to be merged yet.

Reply:
That's great! Could you help link the PR of that code? Thanks

Reply:
https://github.com/openshift/external-dns-operator/pull/50